lib/index.ts

```diff
@@ -56,7 +56,10 @@ export const useStoryblok = async (
   const story: Ref<ISbStoryData> = ref(null);
 
   onMounted(() => {
-    if (story.value && story.value.id) {
+    const storyId = new URL(window.location.href).searchParams.get(
+      "_storyblok"
+    );
+    if (story.value && story.value.id && story.value.id == +storyId) {
```

Comment:
Since we are targeting ESNext, I think you could use an optional chaining operator here.

```
if(story?.value?.id === +storyId) 
```